meta.go

```diff
@@ -1,20 +1,192 @@
 package main
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
+	"io/ioutil"
 	"os"
+	"reflect"
 	"runtime/debug"
+	"strconv"
 
 	"github.com/urfave/cli"
 )
 
 // VERSION gets set by the build script via the LDFLAGS
 var VERSION string
 
-func getMeta(key string, metaSpace string) {
+var mkdirAll = os.MkdirAll
+var stat = os.Stat
+var writeFile = ioutil.WriteFile
+var readFile = ioutil.ReadFile
+var fprintf = fmt.Fprintf
+
+const metaFile = "meta.json"
+
+// Get meta from file based on key
+func getMeta(key string, metaSpace string, output io.Writer) {
+	metaFilePath := metaSpace + "/" + metaFile
+	metaJson, err := readFile(metaFilePath)
+	if err != nil {
+		panic(err)
+	}
+	var metaInterface map[string]interface{}
+	err = json.Unmarshal(metaJson, &metaInterface)
+	if err != nil {
+		panic(err)
+	}
+
+	result := metaInterface[key]
+	switch result.(type) {
+	case map[string]interface{}, []interface{}:
+		resultJson, _ := json.Marshal(result)
+		fprintf(output, "%v", string(resultJson[:]))
```

Comment:
Question: what is the difference between `string(resultJson[:])` and `string(resultJson)`?